torch/nn/modules/module.py

```diff
@@ -600,14 +600,19 @@ def to(self, *args, **kwargs):
         device, dtype, non_blocking, convert_to_format = torch._C._nn._parse_to(*args, **kwargs)
 
         if dtype is not None:
-            if not dtype.is_floating_point:
-                raise TypeError('nn.Module.to only accepts floating point '
+            if not (dtype.is_floating_point or dtype.is_complex):
+                raise TypeError('nn.Module.to only accepts floating point or complex '
                                 'dtypes, but got desired dtype={}'.format(dtype))
+            if dtype.is_complex:
+                warnings.warn(
+                    "Complex modules are a new feature, and some modules might not work as expected "
```

Comment:
I like your version of this message better than the one I suggested.

nit: extra space at the end

Reply:
The extra space at the end of this string is needed so that when it's concatenated with the next one there is a space between words, otherwise the warning message would be `"... as expectedwhen using ..."`

Reply:
Right. But there's a space after the period on the next line. `buffers. ")` -> `buffers.")`

Reply:
this warning message looks good! I think we could also go an extra mile and add "please report an issue on PyTorch if something doesn't work as expected"